albatross/evaluation/evaluate.h

```diff
@@ -68,7 +80,24 @@ negative_log_likelihood(const JointDistribution &prediction,
   return albatross::negative_log_likelihood(mean, covariance);
 }
 
-} // namespace evaluation_metrics
+static inline double
+negative_log_likelihood(const MarginalDistribution &prediction,
+                        const MarginalDistribution &truth) {
+  const Eigen::VectorXd mean = prediction.mean - truth.mean;
+  Eigen::VectorXd variance(prediction.covariance.diagonal());
+  if (truth.has_covariance()) {
```

Comment:
Can we have a Marginal where `has_covariance()` would be false? By definition, isn't a marginal distribution a mean and a diagonal-only covariance?

Reply:
I think we're *very* close to being able to make it so a `Distribution` by definition has a covariance.  But at the moment we support it to allow for things like a `target` used to fit a model which is mean only.

I might be able to do away with `has_covariance()` entirely if we default to zero variance.  I'll see how hard that'd be.

Reply:
My question is more about how the function arguments are a `MarginalDistribution`, and not a `Distribution`.

Reply:
If you create a marginal distribution with a mean only `has_covariance()` will return `false`.